app/services/process_submission_service.rb

```diff
@@ -24,6 +24,22 @@ def perform # rubocop:disable Metrics/AbcSize
             key: action.fetch(:encryption_key)
           )
         ).execute(submission: payload_service.submission, service_slug: submission.service_slug)
+      when 'email'
+        EmailOutputService.new(
+          pdf_generator: SaveTempPdf.new(
+            generate_pdf_content_service: GeneratePdfContent.new(
+              pdf_api_gateway: pdf_gateway(submission.service_slug),
+              payload: pdf_detail.with_indifferent_access
+            ),
+            tmp_file_gateway: Tempfile
+          ),
+          attachment_downloader: DownloadService.new(
+            attachments: payload_service.attachments,
+            target_dir: nil,
+            submission: submission
+          ),
+          emailer: EmailService.new()
+        ).excute(instructions: action)
```

Comment:
This is the intended API. Purely pseudo code.

Reply:
The issue with this approach is that we will be generating pdf's and downloading attachments multiple times if for instance the user and the form owner need a pdf. This duplicates downloading.

The other option is downloading all the attachments and creating the pdf in advance, then iterating over the actions. This duplicates logic. Eg:
```
    generated_attachments = DownloadService.new().execute if payload_service.actions.map{|a| a["include_attachments"]}
    pdf = generate_pdf_file if payload_service.actions.map{ |a| a["include_pdf"]}
    attachments = {
      pdf: pdf,
      attachments: generated_attachments
    }

    payload_service.actions.each do |action|
      case action.fetch(:type)
      when 'email'
        EmailOutputService.new(
          emailer: EmailService.new()
        ).excute(instructions: action, attachments: attachments, pdf: pdf)
      when 'json'
        JsonWebhookService.new(
...
```